core/src/main/java/fr/sncf/osrd/api/InfraCacheStatusEndpoint.java

```diff
@@ -44,15 +44,23 @@ public Response act(Request req) throws IOException {
         // Parse request input
         try {
             var body = new RqPrint(req).printBody();
+            String infra;
+            if (!body.equals("")) {
+                infra = adapterRequest.fromJson(body).infra;
 
-            if (body.equals("")) {
+                if (infra == null) {
+                    infraManager.forEach((infraId, infraCacheEntry) -> {
+                        res.put(infraId, SerializedInfraCache.from(infraCacheEntry));
+                    });
+                } else {
+                    var request = adapterRequest.fromJson(body);
+                    var infraCacheEntry = infraManager.getInfraCache(request.infra);
+                    res.put(request.infra, SerializedInfraCache.from(infraCacheEntry));
+                }
+            } else {
```

Comment:
You could avoid the redundancy by doing
```suggestion
            }
            if (infra != null) {
                var request = adapterRequest.fromJson(body);
                var infraCacheEntry = infraManager.getInfraCache(request.infra);
                res.put(request.infra, SerializedInfraCache.from(infraCacheEntry));
            } else {
```